Web/react-spa/src/features/products/productsSlice.ts

```diff
@@ -0,0 +1,49 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "app/store";
+import Product from "./product";
+import ProductAPI from "./productAPI";
+
+export interface productsState {
```

Comment:
usually, interfaces are kept separately from places they are used, in types.d.ts files